src/test/java/chessgame/state/StateTest.java

```diff
@@ -0,0 +1,151 @@
+package chessgame.state;
+
+import chessgame.domain.Board;
+import chessgame.domain.ChessBoardFactory;
+import chessgame.domain.Command;
+import chessgame.domain.state.*;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class StateTest {
+    @Nested
+    @DisplayName("ready 상태 일 경우")
+    class ReadyState {
+        State state = new Ready();
+
+        @Test
+        @DisplayName("start 입력시 상태 테스트")
+        void Should_ChangeWhiteState_When_ReadyState() {
+            Board board = new Board(ChessBoardFactory.create());
+
+            State start = state.run(Command.of("start"), board);
+            assertThat(start).isInstanceOf(White.class);
+        }
+
+        @Test
+        @DisplayName("move 입력시 상태 테스트")
+        void Should_ThrowException_When_ReadyState() {
+            Board board = new Board(ChessBoardFactory.create());
+            assertThatThrownBy(() -> state.run(Command.of("move a2 a4"), board))
+                    .isInstanceOf(IllegalArgumentException.class)
+                    .hasMessage("move를 입력할 수 없습니다.");
+        }
```

Comment:
```suggestion
    @DisplayName("ready 상태 일 경우")
    class ReadyState {
        State state = new Ready();

        @Test
        @DisplayName("start 입력시 white 상태로 전이된다")
        void Should_ChangeWhiteState_When_ReadyState() {
            Board board = new Board(ChessBoardFactory.create());

            State start = state.run(Command.of("start"), board);
            assertThat(start).isInstanceOf(White.class);
        }

        @Test
        @DisplayName("move 입력시 예외가 발생한다")
        void Should_ThrowException_When_ReadyState() {
            Board board = new Board(ChessBoardFactory.create());
            assertThatThrownBy(() -> state.run(Command.of("move a2 a4"), board))
                    .isInstanceOf(IllegalArgumentException.class)
                    .hasMessage("move를 입력할 수 없습니다.");
        }
```

DCI를 적용하여 컨텍스트를 잘 나눠주셨어요 👍
조금 더 나가면 각 테스트의 내부 설명만 읽어도 해당 테스트를 이해할 수 있도록 작성해줘도 좋겠어요!